### PR TITLE
Softmax monte -- masked and non masked variants, both monte and monte_bylpvip

### PR DESCRIPTION
**Pull request by abracodeabra1**
# xlnscpp PR: Align monte softmax with xlns16 (scale + masked)

## Summary

Bring `xlns16_softmax_monte` in line with `xlns16_softmax`, and add `xlns16_softmax_masked_monte`, in both Monte backends (`xlns16monte.cpp` and `xlns16monte_bylpvip.cpp`):

```
softmax_monte(a, scale)        = exp(scale*a[i] - max) / sum_j exp(scale*a[j] - max)
softmax_masked_monte(a, mask)  = same, after applying mask to scaled logits
```

Mask semantics match xlns16: `xlns16_neg_inf` hard-excludes a position, `xlns16_zero` is a no-op, and any other value is an additive bias (e.g. ALiBi). Scaling is skipped when `a[i]` is already `neg_inf` so the sentinel bit pattern is preserved. `c` may alias `a` (in-place).

Monte (`xlns16_add_monte`) is used for every add that appears: mask bias, max-subtraction (`add_monte(c[i], signmask^maxval)`), and the normalization sum via `xlns16_sum_monte`. Both backends keep identical softmax bodies; only `xlns16_add_monte` differs (pure 16-bit MCLNS vs same-noise + `xlns32_add_lpvip`).

## Motivation

Monte softmax previously lacked `scale` and a masked variant.

## Testing

Pure monte (`xlns16_alt` + `xlns16_table`; needs non-ideal path for `monte_modifier`):

```
plain softmax max abs diff vs float ref: 0.003188
dense grid plain softmax (n_samples=2886, vector size 6):
  max absolute error: 0.006146
  max relative error (ref>0.05): 2.4851%
scaled softmax (1/sqrt(64)) max abs diff: 0.000652
masked softmax max abs diff: 0.002802 (excluded slots ~0; sums to ~1)
```

bylpvip backend (`xlns16_ideal` + `xlns32lpvip`): similar spot/grid margins.

## Notes

- Default `scale = xlns16_one` keeps existing 3-arg call sites compiling unchanged.

### CODE REVIEW

**Comment by markgarnold** [comment]
Should use Monte on summation only. Won't make much difference, but extra Monte creates additional non-determism that might make debugging harder
